File: 14.最长公共前缀.cpp
/*
 * @lc app=leetcode.cn id=14 lang=cpp
 *
 * [14] 最长公共前缀
 */

// @lc code=start
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        if(strs.size()==0) return "";
        string re="",a=strs[0],b=strs[strs.size()-1];
        int i=0;
        while(a[i]==b[i] && i<a.length() && i<b.length())
            re += a[i++];
        return re; 
    }
};
// @lc code=end

File: 14.最长公共前缀.java
import java.util.Arrays;

/*
 * @lc app=leetcode.cn id=14 lang=java
 *
 * [14] 最长公共前缀
 */

// @lc code=start
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        if(strs.length==0) return "";
        String a=strs[0],b=strs[strs.length-1];
        StringBuilder re=new StringBuilder();
        int i=0;
        while(i<a.length() && i<b.length() && a.charAt(i)==b.charAt(i))
            re.append(a.charAt(i++));
        return re.toString();
    }
}
// @lc code=end

